### ISSUE TITLE
unable to upgrade cluster: creating new cluster on existing cluster is not supported for self managed clusters

### ISSUE THREAD

**Issue by echel0n** [aws/eks-anywhere#3093]
**What happened**:
attempted to perform a cluster upgrade from 1.22 to 1.23, resulted in `creating new cluster on existing cluster is not supported for self managed clusters` and failing to upgrade

**What you expected to happen**:
cluster upgrade to 1.23

**How to reproduce it (as minimally and precisely as possible)**:
edit the cluster yaml and bump `kubernetesVersion` from 1.22 to 1.23, then perform cluster upgrade eksctl command

**Anything else we need to know?**:
this is a vsphere cluster.

after the cluster upgrade failed, I no longer can attempt further upgrades, I get the following error: `Error: failed to upgrade cluster: failed getting EKS-A cluster to build current cluster Spec: cluster prod not found of custom resource type clusters.anywhere.eks.amazonaws.com`.

`
2022-08-21T14:16:15.415-0700    V5      Error happened during retry     {"error": "executing apply --force: Warning: resource clusters/prod is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.\nWarning: resource vspheredatacenterconfigs/prod is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.\nWarning: resource vspheremachineconfigs/prod-cp is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.\nWarning: resource vspheremachineconfigs/prod-etcd is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.\nWarning: resource vspheremachineconfigs/prod is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.\nWarning: resource fluxconfigs/gitlab is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.\nerror: error when applying patch:\n{\"apiVersion\":\"anywhere.eks.amazonaws.com/v1alpha1\",\"kind\":\"Cluster\",\"metadata\":{\"annotations\":{\"kubectl.kubernetes.io/last-applied-configuration\":\"{\\\"apiVersion\\\":\\\"anywhere.eks.amazonaws.com/v1alpha1\\\",\\\"kind\\\":\\\"Cluster\\\",\\\"metadata\\\":{\\\"annotations\\\":{},\\\"name\\\":\\\"prod\\\",\\\"namespace\\\":\\\"default\\\"},\\\"spec\\\":{\\\"bundlesRef\\\":{\\\"apiVersion\\\":\\\"anywhere.eks.amazonaws.com/v1alpha1\\\",\\\"name\\\":\\\"bundles-12\\\",\\\"namespace\\\":\\\"eksa-system\\\"},\\\"clusterNetwork\\\":{\\\"cniConfig\\\":{\\\"cilium\\\":{}},\\\"pods\\\":{\\\"cidrBlocks\\\":[\\\"10.168.0.0/16\\\"]},\\\"services\\\":{\\\"cidrBlocks\\\":[\\\"10.96.0.0/12\\\"]}},\\\"controlPlaneConfiguration\\\":{\\\"count\\\":2,\\\"endpoint\\\":{\\\"host\\\":\\\"192.168.3.10\\\"},\\\"machineGroupRef\\\":{\\\"kind\\\":\\\"VSphereMachineConfig\\\",\\\"name\\\":\\\"prod-cp\\\"}},\\\"datacenterRef\\\":{\\\"kind\\\":\\\"VSphereDatacenterConfig\\\",\\\"name\\\":\\\"prod\\\"},\\\"externalEtcdConfiguration\\\":{\\\"count\\\":3,\\\"machineGroupRef\\\":{\\\"kind\\\":\\\"VSphereMachineConfig\\\",\\\"name\\\":\\\"prod-etcd\\\"}},\\\"gitOpsRef\\\":{\\\"kind\\\":\\\"FluxConfig\\\",\\\"name\\\":\\\"gitlab\\\"},\\\"kubernetesVersion\\\":\\\"1.23\\\",\\\"managementCluster\\\":{\\\"name\\\":\\\"prod\\\"},\\\"workerNodeGroupConfigurations\\\":[{\\\"count\\\":4,\\\"machineGroupRef\\\":{\\\"kind\\\":\\\"VSphereMachineConfig\\\",\\\"name\\\":\\\"prod\\\"},\\\"name\\\":\\\"md-0\\\"}]}}\\n\"},\"name\":\"prod\",\"namespace\":\"default\"},\"spec\":{\"bundlesRef\":{\"apiVersion\":\"anywhere.eks.amazonaws.com/v1alpha1\",\"name\":\"bundles-12\",\"namespace\":\"eksa-system\"},\"clusterNetwork\":{\"cniConfig\":{\"cilium\":{}},\"pods\":{\"cidrBlocks\":[\"10.168.0.0/16\"]},\"services\":{\"cidrBlocks\":[\"10.96.0.0/12\"]}},\"controlPlaneConfiguration\":{\"count\":2,\"endpoint\":{\"host\":\"192.168.3.10\"},\"machineGroupRef\":{\"kind\":\"VSphereMachineConfig\",\"name\":\"prod-cp\"}},\"datacenterRef\":{\"kind\":\"VSphereDatacenterConfig\",\"name\":\"prod\"},\"externalEtcdConfiguration\":{\"count\":3,\"machineGroupRef\":{\"kind\":\"VSphereMachineConfig\",\"name\":\"prod-etcd\"}},\"gitOpsRef\":{\"kind\":\"FluxConfig\",\"name\":\"gitlab\"},\"kubernetesVersion\":\"1.23\",\"managementCluster\":{\"name\":\"prod\"},\"workerNodeGroupConfigurations\":[{\"count\":4,\"machineGroupRef\":{\"kind\":\"VSphereMachineConfig\",\"name\":\"prod\"},\"name\":\"md-0\"}]}}\n\nto:\nResource: \"anywhere.eks.amazonaws.com/v1alpha1, Resource=clusters\", GroupVersionKind: \"anywhere.eks.amazonaws.com/v1alpha1, Kind=Cluster\"\nName: \"prod\", Namespace: \"default\"\nfor: \"STDIN\": An error occurred force-replacing the existing object with the newly provided one:\n\nadmission webhook \"validation.cluster.anywhere.amazonaws.com\" denied the request: creating new cluster on existing cluster is not supported for self managed clusters.\n\nAdditionally, an error occurred attempting to restore the original object:\n\nadmission webhook \"validation.cluster.anywhere.amazonaws.com\" denied the request: creating new cluster on existing cluster is not supported for self managed clusters\n", "retries": 1}
`

**Environment**:
- EKS Anywhere Release: 0.11.0
- EKS Distro Release: 1.22 -> 1.23

**Comment by mitalipaygude** [comment]
Hello @echel0n,
 
Thank you for reporting this issue. If you provide the below information to us, we can help debug the issue better.

Can you please check if the eksa cluster has the pause annotation created and set to true using kubectl get? 
Can you also attach your cluster spec to this issue - the one you used to create, and then the one you used to upgrade? Was the kubernetes version the only field that changed during the upgrade?

Which eksctl anywhere version did you use to create the original cluster? Was it v0.10.1?

**Comment by echel0n** [comment]
I'm actually unable to check the eksa cluster for the pause annotation as after it failed to update the cluster it actually deleted it or lost it, I had to modify the webhook admission just so I could re-create it, was very strange.

Original cluster was created with `v0.10.1` and then upgrade was performed with `v0.11.0`

Only thing changed was the kubernetes version.

Below is the config I used prior to the upgrade:
```
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: Cluster
metadata:
  name: prod
  namespace: default
spec:
  bundlesRef:
    apiVersion: anywhere.eks.amazonaws.com/v1alpha1
    name: bundles-12
    namespace: eksa-system
  clusterNetwork:
    cniConfig:
      cilium: {}
    pods:
      cidrBlocks:
      - 10.168.0.0/16
    services:
      cidrBlocks:
      - 10.96.0.0/12
  controlPlaneConfiguration:
    count: 2
    endpoint:
      host: 192.168.3.10
    machineGroupRef:
      kind: VSphereMachineConfig
      name: prod-cp
  datacenterRef:
    kind: VSphereDatacenterConfig
    name: prod
  externalEtcdConfiguration:
    count: 3
    machineGroupRef:
      kind: VSphereMachineConfig
      name: prod-etcd
  gitOpsRef:
    kind: FluxConfig
    name: gitlab
  kubernetesVersion: "1.22"
  managementCluster:
    name: prod
  workerNodeGroupConfigurations:
  - count: 4
    machineGroupRef:
      kind: VSphereMachineConfig
      name: prod
    name: md-0

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereDatacenterConfig
metadata:
  name: prod
  namespace: default
spec:
  datacenter: Dark Systems Datacenter
  insecure: true
  network: /Dark Systems Datacenter/network/DSwitch-10GB-EKS
  server: vcenter.vsphere.darksystems.ca
  thumbprint: ""

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereMachineConfig
metadata:
  annotations:
    anywhere.eks.amazonaws.com/control-plane: "true"
  name: prod-cp
  namespace: default
spec:
  datastore: /Dark Systems Datacenter/datastore/vsanDatastore
  diskGiB: 25
  folder: /Dark Systems Datacenter/vm/EKS Anywhere
  memoryMiB: 8192
  numCPUs: 2
  osFamily: bottlerocket
  resourcePool: /Dark Systems Datacenter/host/Cluster/Resources
  template: /Dark Systems Datacenter/vm/Templates/bottlerocket-vmware-k8s-1.22-x86_64-1.8.0-a6233c22
  users:
  - name: ec2-user
    sshAuthorizedKeys:
    - ""

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereMachineConfig
metadata:
  annotations:
    anywhere.eks.amazonaws.com/etcd: "true"
  name: prod-etcd
  namespace: default
spec:
  datastore: /Dark Systems Datacenter/datastore/vsanDatastore
  diskGiB: 25
  folder: /Dark Systems Datacenter/vm/EKS Anywhere
  memoryMiB: 8192
  numCPUs: 2
  osFamily: bottlerocket
  resourcePool: /Dark Systems Datacenter/host/Cluster/Resources
  template: /Dark Systems Datacenter/vm/Templates/bottlerocket-vmware-k8s-1.22-x86_64-1.8.0-a6233c22
  users:
  - name: ec2-user
    sshAuthorizedKeys:
    - ""

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereMachineConfig
metadata:
  name: prod
  namespace: default
spec:
  datastore: /Dark Systems Datacenter/datastore/vsanDatastore
  diskGiB: 50
  folder: /Dark Systems Datacenter/vm/EKS Anywhere
  memoryMiB: 16384
  numCPUs: 16
  osFamily: bottlerocket
  resourcePool: /Dark Systems Datacenter/host/Cluster/Resources
  template: /Dark Systems Datacenter/vm/Templates/bottlerocket-vmware-k8s-1.22-x86_64-1.8.0-a6233c22
  users:
  - name: ec2-user
    sshAuthorizedKeys:
    - ""

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: FluxConfig
metadata:
  name: gitlab
  namespace: default
spec:
  branch: master
  clusterConfigPath: clusters/prod
  git:
    repositoryUrl: ssh://git@***.com/***/k8s-cluster.git
    sshKeyAlgorithm: ecdsa
  systemNamespace: flux-system

---
```
Below is the config I used to upgrade with:
```
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: Cluster
metadata:
  name: prod
  namespace: default
spec:
  bundlesRef:
    apiVersion: anywhere.eks.amazonaws.com/v1alpha1
    name: bundles-12
    namespace: eksa-system
  clusterNetwork:
    cniConfig:
      cilium: {}
    pods:
      cidrBlocks:
      - 10.168.0.0/16
    services:
      cidrBlocks:
      - 10.96.0.0/12
  controlPlaneConfiguration:
    count: 2
    endpoint:
      host: 192.168.3.10
    machineGroupRef:
      kind: VSphereMachineConfig
      name: prod-cp
  datacenterRef:
    kind: VSphereDatacenterConfig
    name: prod
  externalEtcdConfiguration:
    count: 3
    machineGroupRef:
      kind: VSphereMachineConfig
      name: prod-etcd
  gitOpsRef:
    kind: FluxConfig
    name: gitlab
  kubernetesVersion: "1.23"
  managementCluster:
    name: prod
  workerNodeGroupConfigurations:
  - count: 4
    machineGroupRef:
      kind: VSphereMachineConfig
      name: prod
    name: md-0

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereDatacenterConfig
metadata:
  name: prod
  namespace: default
spec:
  datacenter: Dark Systems Datacenter
  insecure: true
  network: /Dark Systems Datacenter/network/DSwitch-10GB-EKS
  server: vcenter.vsphere.darksystems.ca
  thumbprint: ""

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereMachineConfig
metadata:
  annotations:
    anywhere.eks.amazonaws.com/control-plane: "true"
  name: prod-cp
  namespace: default
spec:
  datastore: /Dark Systems Datacenter/datastore/vsanDatastore
  diskGiB: 25
  folder: /Dark Systems Datacenter/vm/EKS Anywhere
  memoryMiB: 8192
  numCPUs: 2
  osFamily: bottlerocket
  resourcePool: /Dark Systems Datacenter/host/Cluster/Resources
  template: /Dark Systems Datacenter/vm/Templates/bottlerocket-vmware-k8s-1.23-x86_64-1.9.0-159e4ced
  users:
  - name: ec2-user
    sshAuthorizedKeys:
    - ""

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereMachineConfig
metadata:
  annotations:
    anywhere.eks.amazonaws.com/etcd: "true"
  name: prod-etcd
  namespace: default
spec:
  datastore: /Dark Systems Datacenter/datastore/vsanDatastore
  diskGiB: 25
  folder: /Dark Systems Datacenter/vm/EKS Anywhere
  memoryMiB: 8192
  numCPUs: 2
  osFamily: bottlerocket
  resourcePool: /Dark Systems Datacenter/host/Cluster/Resources
  template: /Dark Systems Datacenter/vm/Templates/bottlerocket-vmware-k8s-1.23-x86_64-1.9.0-159e4ced
  users:
  - name: ec2-user
    sshAuthorizedKeys:
    - ""

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereMachineConfig
metadata:
  name: prod
  namespace: default
spec:
  datastore: /Dark Systems Datacenter/datastore/vsanDatastore
  diskGiB: 50
  folder: /Dark Systems Datacenter/vm/EKS Anywhere
  memoryMiB: 16384
  numCPUs: 16
  osFamily: bottlerocket
  resourcePool: /Dark Systems Datacenter/host/Cluster/Resources
  template: /Dark Systems Datacenter/vm/Templates/bottlerocket-vmware-k8s-1.23-x86_64-1.9.0-159e4ced
  users:
  - name: ec2-user
    sshAuthorizedKeys:
    - ""

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: FluxConfig
metadata:
  name: gitlab
  namespace: default
spec:
  branch: master
  clusterConfigPath: clusters/prod
  git:
    repositoryUrl: ssh://git@***.com/***/k8s-cluster.git
    sshKeyAlgorithm: ecdsa
  systemNamespace: flux-system

---
```

**Comment by lewisdiamond** [comment]
Fixed by #3108
